### ISSUE TITLE
False positive warning about failing to cast to callable when protected method is inherited

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#1427]
This happens when protectedMethod is declared in the base class, but not when it's in the same subclass as where it's used.

Expected: no error
Observed:

```
ERROR: InvalidArgument - src/fromcallable.php:11:38 - Argument 1 of fromCallable expects callable, array{0:Example, 1:string(protectedMethod)} provided
        return Closure::fromCallable([$this, 'protectedMethod']);
```

```php
<?php

class Base  {
    protected function protectedMethod() : void {
        echo "In protected method\n";
    }
}

class Example extends Base {
    public function test() : Closure {
        return Closure::fromCallable([$this, 'protectedMethod']);
    }
}
```
